From: Bart Van Assche <bvanassche@acm.org>
Date: Tue, 19 Apr 2022 15:58:05 -0700
Subject: scsi: ufs: qcom: Fix ufs_qcom_resume()
Git-commit: bee40dc167da159ea5b939c074e1da258610a3d6
Patch-mainline: v5.19-rc1
References: git-fixes

Clearing hba->is_sys_suspended if ufs_qcom_resume() succeeds is wrong. That
variable must only be cleared if all actions involved in a resume succeed.
Hence remove the statement that clears hba->is_sys_suspended from
ufs_qcom_resume().

[lduncan: hand applied and refreshed.]

Link: https://lore.kernel.org/r/20220419225811.4127248-23-bvanassche@acm.org
Fixes: 81c0fc51b7a7 ("ufs-qcom: add support for Qualcomm Technologies Inc platforms")
Tested-by: Bean Huo <beanhuo@micron.com>
Reviewed-by: Bjorn Andersson <bjorn.andersson@linaro.org>
Reviewed-by: Bean Huo <beanhuo@micron.com>
Signed-off-by: Bart Van Assche <bvanassche@acm.org>
Signed-off-by: Martin K. Petersen <martin.petersen@oracle.com>
Acked-by: Lee Duncan <lduncan@suse.com>
---
 drivers/scsi/ufs/ufs-qcom.c |   11 ++---------
 1 file changed, 2 insertions(+), 9 deletions(-)

--- a/drivers/scsi/ufs/ufs-qcom.c
+++ b/drivers/scsi/ufs/ufs-qcom.c
@@ -602,17 +602,10 @@ static int ufs_qcom_resume(struct ufs_hb
 	if (err) {
 		dev_err(hba->dev, "%s: failed enabling regs, err = %d\n",
 			__func__, err);
-		goto out;
+		return err;
 	}
 
-	err = ufs_qcom_enable_lane_clks(host);
-	if (err)
-		goto out;
-
-	hba->is_sys_suspended = false;
-
-out:
-	return err;
+	return ufs_qcom_enable_lane_clks(host);
 }
 
 struct ufs_qcom_dev_params {
